packages/calcite-components/src/components/date-picker-month-header/date-picker-month-header.tsx

```diff
@@ -119,68 +149,130 @@ export class DatePickerMonthHeader {
+          {this.position !== "start" && this.renderChevron("right")}
         </div>
-        <a
-          aria-disabled={`${this.nextMonthDate.getMonth() === activeMonth}`}
-          aria-label={messages.nextMonth}
-          class={CSS.chevron}
-          href="#"
-          onClick={this.nextMonthClick}
-          onKeyDown={this.nextMonthKeydown}
-          role="button"
-          tabindex={this.nextMonthDate.getMonth() === activeMonth ? -1 : 0}
-        >
-          <calcite-icon flip-rtl icon={ICON.chevronRight} scale={getIconScale(this.scale)} />
-        </a>
       </Fragment>
     );
   }
 
+  private renderMonthYearContainer(reverse: boolean): VNode {
+    return reverse ? (
+      <Fragment>
+        {this.renderYearInput()}
+        {this.renderMonthPicker()}
+      </Fragment>
+    ) : (
+      <Fragment>
+        {this.renderMonthPicker()}
+        {this.renderYearInput()}
+      </Fragment>
+    );
+  }
+
+  private renderMonthPicker(): VNode {
+    const activeMonth = this.activeDate.getMonth();
+    const monthData = this.localeData.months[this.monthStyle];
+    return (
+      <calcite-select
+        class={CSS.monthPicker}
+        label={this.messages.monthMenu}
+        onCalciteSelectChange={this.handleMonthChange}
+        onKeyDown={this.handleKeyDown}
+        ref={(el) => (this.monthPickerEl = el)}
+        width="auto"
+      >
+        {monthData.map((month: string, index: number) => {
+          return (
+            <calcite-option
+              disabled={!this.isMonthInRange(index)}
+              selected={index === activeMonth}
+              value={month}
+            >
+              {month}
+            </calcite-option>
+          );
+        })}
+      </calcite-select>
+    );
+  }
+
+  private renderYearInput(): VNode {
+    const suffix = this.localeData.year?.suffix;
+    const localizedYear = this.formatCalendarYear(this.activeDate.getFullYear());
+    return (
+      <span class={CSS.yearContainer}>
+        <input
+          aria-label={this.messages.year}
+          class={{
+            year: true,
+          }}
+          inputmode="numeric"
+          maxlength="4"
+          minlength="1"
+          onChange={this.onYearChange}
+          onInput={this.onYearInput}
+          onKeyDown={this.onYearKey}
+          pattern="\d*"
+          ref={(el) => (this.yearInputEl = el)}
+          type="text"
+          value={localizedYear}
+        />
+        {suffix && <span class={CSS.suffix}>{suffix}</span>}
+      </span>
+    );
+  }
+
+  private renderChevron(direction: "left" | "right"): VNode {
+    const isDirectionRight = direction === "right";
+    const isDisabled =
+      hasSameMonthAndYear(
+        isDirectionRight ? this.nextMonthDate : this.prevMonthDate,
+        this.activeDate,
+      ) || !inRange(this.activeDate, this.min, this.max);
+
+    return (
+      <calcite-action
+        alignment="center"
+        aria-disabled={`${isDisabled}`}
```

Comment:
Can you use `toAriaBoolean` for ARIA boolean attrs? Applies to other ARIA booleans.